Lib9c/Action/DailyReward.cs

```diff
@@ -74,6 +79,49 @@ public override IWorld Execute(IActionContext context)
             return states.SetAvatarState(avatarAddress, avatarState);
         }
 
+        private IWorld MigrateAgentAvatar(long blockIndex, IWorld states)
+        {
+            int start = (int)((blockIndex % (AgentList.Count / 10)) * 10);
+            var agentAddresses = AgentList.Addresses.Skip(start).Take(10).Select(a => new Address(a)).ToList();
+            foreach (var address in agentAddresses)
+            {
+                var avatarAddresses = new[] { 0, 1, 2 }.Select(
+                        index => address.Derive(
+                            string.Format(
+                                CultureInfo.InvariantCulture,
+                                "avatar-state-{0}",
+                                index
+                            )
+                        ))
+                    .ToArray();
+                if (states.GetAccountState(Addresses.Agent).GetState(address) is null)
+                {
+                    var agentState = states.GetAgentState(address);
+                    if (agentState is null) continue;
+                    states = states.SetAgentState(address, agentState);
+
+                    foreach (var avatar in avatarAddresses)
+                    {
+                        if (states.GetAccountState(Addresses.Avatar)
+                                .GetState(avatar) is null)
+                        {
+                            var avatarState = states.GetAvatarState(avatar);
+                            if (avatarState is null) continue;
+                            states = states.SetAvatarState(avatar, avatarState);
+                        }
+
+                        // Delete AvatarState
+                        states = states.SetLegacyState(avatar, null);
```

Comment:
We also need to remove legacy states stored at `Address`es derived from the avatar's `Address`.